packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.js

```diff
@@ -1,35 +1,64 @@
-import algoliaClient from 'algoliasearch/lite';
 import createInstantSearchManager from '../createInstantSearchManager';
 
 jest.useFakeTimers();
 
-const defaultResponse = () => ({
-  results: [
-    {
-      params: 'query=&hitsPerPage=10&page=0&facets=%5B%5D&tagFilters=',
-      page: 0,
-      hits: [],
-      hitsPerPage: 10,
-      nbPages: 0,
-      processingTimeMS: 4,
-      query: '',
-      nbHits: 0,
-      index: 'index',
-    },
-  ],
+const createSearchClient = () => ({
+  search: jest.fn(() =>
+    Promise.resolve({
+      results: [
+        {
+          params: 'query=&hitsPerPage=10&page=0&facets=%5B%5D&tagFilters=',
+          page: 0,
+          hits: [],
+          hitsPerPage: 10,
+          nbPages: 0,
+          processingTimeMS: 4,
+          query: '',
+          nbHits: 0,
+          index: 'index',
+        },
+      ],
+    })
+  ),
 });
 
+const createControlledSearchClient = () => {
+  const searchResultsPromises = [];
+
+  return {
+    search: jest.fn(() => {
+      const results = Promise.resolve({
+        results: [
+          {
+            params: 'query=&hitsPerPage=10&page=0&facets=%5B%5D&tagFilters=',
+            page: 0,
+            hits: [],
+            hitsPerPage: 10,
+            nbPages: 0,
+            processingTimeMS: 4,
+            query: '',
+            nbHits: 0,
+            index: 'index',
+          },
+        ],
+      });
+
+      searchResultsPromises.push(results);
```

Comment:
It seems you kept this variable from `makeManagedClient` but it's not exposed anymore and you don't use it in your tests.
Is it really needed?